src/aircraft_cmd.cpp

```diff
@@ -2147,18 +2149,7 @@ void UpdateAirplanesOnNewStation(const Station *st)
 		if (!v->IsNormalAircraft() || v->targetairport != st->index) continue;
 		assert(v->state == FLYING);
 
-		Order *o = &v->current_order;
-		/* The aircraft is heading to a hangar, but the new station doesn't have one,
-		 * or the aircraft can't land on the new station. Cancel current order. */
-		if (o->IsType(OT_GOTO_DEPOT) && !(o->GetDepotOrderType() & ODTFB_PART_OF_ORDERS) && o->GetDestination() == st->index &&
-				(!st->airport.HasHangar() || !CanVehicleUseStation(v, st))) {
-			o->MakeDummy();
-			SetWindowWidgetDirty(WC_VEHICLE_VIEW, v->index, WID_VV_START_STOP);
-		}
 		v->pos = v->previous_pos = AircraftGetEntryPoint(v, ap, rotation);
 		UpdateAircraftCache(v);
 	}
-
-	/* Heliports don't have a hangar. Invalidate all go to hangar orders from all aircraft. */
-	if (!st->airport.HasHangar()) RemoveOrderFromAllVehicles(OT_GOTO_DEPOT, st->index, true);
```

Comment:
Same as above, help us (me) understand why this isn't needed anymore.

Reply:
Old goto hangar orders stored the StationID. It was possible to create a goto hangar order to a station having a hangar, demolish the airport and rebuilt it with another airport that had no hangar. Thus, when rebuilding an airport that had no hangar, all goto hangar orders were to be removed.

Adding a depot struct to airports when they have a hangar, allows hangar orders to store a DepotID. Most of the special cases you had to deal with hangar orders, can now be handled by the standard way (rail, road and water transport code).

With this commit, old goto depot orders to airports that don't have a hangar are removed on loading a game.

Also, when an airport is removed, their goto depot orders are also removed. This could be a problem, as the immediate removal of these orders is not always desired (rebuilding an airport). In subsequent commits, when a depot is deleted, it will stay for a while as a ghost (to keep orders associated to it) and eventually will be removed forever (behaving in a similar way to station removal).

Reply:
Ah, so instead this is handled by the change to `RemoveOrderFromAllVehicles()`?

Reply:
Yes. It is called during actions such as removing an airport, which deletes the associated depot, which calls RemoveOrderFromAllVehicles() in Depot::~Depot().